content/300-guides/200-deployment/201-serverless/400-deploy-to-aws-lambda.mdx

```diff
@@ -1,301 +1,217 @@
-
-The repository contains an `.env.example` example file to help with this.
+This is necessary because the runtime used in the development and when your application is deployed differ. Add the [`binaryTarget`](/reference/api-reference/prisma-schema-reference#binarytargets-options) to make the compatible Prisma engine file available.
 
-### Steps
+### Lambda functions with arm64 architectures
 
-1. Enter the following command to copy `.env.example` to a new file, `.env`:
+Lambda functions that use [arm64 architectures (AWS Graviton2 processor)](https://docs.aws.amazon.com/lambda/latest/dg/foundation-arch.html#foundation-arch-adv) must use an `arm64` precompiled engine file.
 
-   ```terminal
-   cp .env.example .env
-   ```
+In the `generator` block of your `schema.prisma` file, add the following:
 
-2. Open `.env` and set the `DATABASE_URL` to your database credentials. The format is as follows:
+```prisma file=schema.prisma
+binaryTargets = ["native", "linux-arm64-openssl-1.0.x"]
+```
 
-   ```file
-   DATABASE_URL="postgres://__USER__:__PASSWORD__@__HOST__/__DATABASE__"
-   ```
+## Loading environment variables via a <inlinecode>.env</inlinecode> file
 
-   Replace the uppercase placeholders with your database credentials. For example:
+Your functions will need the `DATABASE_URL` environment variable to access the database. The `serverless-dotenv-plugin` will allow you to use your `.env` file in your deployments.
 
-   ```file
-   DATABASE_URL="postgres://janedoe:randompassword@yourpostgres.supabase.co:6543/yourdbname"
-   ```
+First, make sure that the plugin is installed
 
-   > **Note:** If your connection URL contains any special characters, then you must [percentage encode them](/reference/database-reference/connection-urls#special-characters).
+```sh
+npm install -D serverless-dotenv-plugin
+```
 
-> **Note:** It's best practice to keep secret information, such as `DATABASE_URL`, out of your git repository. To do this, add your `.env` file to `.gitignore`. This guide copies the source and does not create a repository, so you only need to update `.gitignore` if you initialize a repository.
+Then, add `serverless-dotenv-plugin` to your list of plugins in `serverless.yml`
 
-## 3. Push the Prisma schema to the database
+```code file=serverless.yml no-copy
+plugins:
+  - serverless-dotenv-plugin
+```
 
-`prisma db push` takes the models and attributes that you define in `schema.prisma` and pushes them to the database.
+The environment variables in your `.env` file will now be automatically loaded on package or deployment.
 
-### Steps
+```sh no-copy highlight=3-4;normal
+❯ serverless package
+Running "serverless" from node_modules
+DOTENV: Loading environment variables from .env:
+         - DATABASE_URL
 
-```terminal
-npx prisma db push
+Packaging deployment-example-sls for stage dev (us-east-1)
+.
+.
+.
 ```
```

Comment:
We could use the `CodeWithResult` component here to separate the command and its output. (Same applies for the Netlify deployment guide)

https://www.prisma.io/docs/about/prisma-docs/docs-components/mdx-examples#code-with-output